apps/src/code-studio/pd/application/facilitator1920/Section3ExperienceAndCommitments.jsx

```diff
@@ -0,0 +1,295 @@
+import React from 'react';
+import {FormGroup} from "react-bootstrap";
+import LabeledFormComponent from "../../form_components/LabeledFormComponent";
+import {
+  PageLabels,
+  SectionHeaders,
+  TextFields
+} from '@cdo/apps/generated/pd/facilitator1920ApplicationConstants';
+import {CSF, CSD, CSP} from '../ApplicationConstants';
+
+const PARTNER_WORKSHOPS_API_ENDPOINT = '/api/v1/pd/regional_partner_workshops/find?';
+
+export default class Section3ExperienceAndCommitments extends LabeledFormComponent {
+  static labels = PageLabels.section3ExperienceAndCommitments;
+
+  static associatedFields = [
+    ...Object.keys(PageLabels.section3ExperienceAndCommitments),
+    "regionalPartnerId"
+  ];
+
+  state = {
+    loadingFitWorkshops: true,
+    loadingSummerWorkshops: false,
+    partner: null,
+    loadError: false
+  };
+
+  componentDidMount() {
+    this.loadFitWorkshops();
+
+    if ([CSD, CSP].includes(this.props.data.program)) {
+      this.loadSummerWorkshops();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.loadFitWorkshopsRequest) {
+      this.loadFitWorkshopsRequest.abort();
+    }
+    if (this.loadSummerWorkshopsRequest) {
+      this.loadSummerWorkshopsRequest.abort();
+    }
+  }
+
+  loadFitWorkshops() {
+    const queryParams = {
+      course: this.props.data.program,
+      subject: 'Code.org Facilitator Weekend',
+      zip_code: this.props.data.zipCode,
+      state: this.props.data.state
+    };
+
+    const url = PARTNER_WORKSHOPS_API_ENDPOINT + $.param(queryParams);
+    this.loadFitWorkshopsRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done((data) => {
+      this.loadFitWorkshopsRequest = null;
+
+      this.handleChange({
+        regionalPartnerId: data.id,
+        fitWorkshops: data.workshops,
+        regionalPartnerName: data.name
+      });
+
+      this.setState({
+        loadingFitWorkshops: false
+      });
+    }).error(() => {
+      this.setState({
+        loadingFitWorkshops: false,
+        loadError: true
+      });
+    });
+  }
+
+  loadSummerWorkshops() {
+    this.setState({loadingSummerWorkshops: true});
+
+    const queryParams = {
+      course: this.props.data.program,
+      subject: '5-day Summer',
+      zip_code: this.props.data.zipCode,
+      state: this.props.data.state
+    };
+
+    const url = PARTNER_WORKSHOPS_API_ENDPOINT + $.param(queryParams);
+    this.loadSummerWorkshopsRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done((data) => {
+      this.loadSummerWorkshopsRequest = null;
+
+      this.handleChange({
+        regionalPartnerId: data.id,
+        summerWorkshops: data.workshops,
+        regionalPartnerName: data.name
+      });
+
+      this.setState({
+        loadingSummerWorkshops: false
+      });
+    }).error(() => {
+      this.setState({
+        loadingSummerWorkshops: false,
+        loadError: true
+      });
+    });
+  }
+
+  renderCsdCspWhichFitWeekend() {
```

Comment:
Bit confusing from the title what is happening here. Suggestions for rename:
renderCsdCspFitWeekendSelector
renderCsdCspFitWeekendOptions
renderFitWeekendOptions

Reply:
It renders the question whose label key is csdCspWhichFitWeekend, which is the naming convention I'm using throughout the application. It's not a different kind of question in the ui (just checkboxes), but we are dynamically creating the options based on the workshops that are in the workshop dashboard at the time the application is being filled out. I can add a comment to explain the dynamic options, if that would help?